dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/data/UploadWorkerTest.kt

```diff
@@ -0,0 +1,160 @@
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class UploadWorkerTest {
+
+    lateinit var testedWorker: Worker
+
+    @Mock
+    lateinit var mockContext: Context
+    @Mock
+    lateinit var mockLogStrategy: LogStrategy
+    @Mock
+    lateinit var mockLogReader: Reader
+    @Mock
+    lateinit var mockLogUploader: LogUploader
+
+    @Forgery
+    lateinit var fakeWorkerParameters: WorkerParameters
+
+    @BeforeEach
+    fun `set up`() {
+        whenever(mockLogStrategy.getLogReader()) doReturn mockLogReader
+
+        mockContext = mockContext()
+        Datadog.initialize(mockContext, "<CLIENT_TOKEN>")
+
+        Datadog.setFieldValue("logStrategy", mockLogStrategy)
+        Datadog.setFieldValue("uploader", mockLogUploader)
+
+        testedWorker = UploadWorker(mockContext, fakeWorkerParameters)
+    }
+
+    @AfterEach
+    fun `tear down`() {
+        Datadog.invokeMethod("stop")
+    }
+
+    @Test
+    fun `doWork single batch Success`(
+        @Forgery batch: Batch
+    ) {
+        whenever(mockLogReader.readNextBatch()).doReturn(batch, null)
+        whenever(mockLogUploader.upload(batch.data)) doReturn LogUploadStatus.SUCCESS
+
+        val result = testedWorker.doWork()
+
+        verify(mockLogReader).dropBatch(batch.id)
+        assertThat(result)
+            .isEqualTo(ListenableWorker.Result.success())
+    }
+
+    @Test
+    fun `doWork single batch Failure`(
+        @Forgery batch: Batch,
+        forge: Forge
+    ) {
+        val status = forge.aValueFrom(
+            LogUploadStatus::class.java,
+            exclude = listOf(LogUploadStatus.SUCCESS)
+        )
+        whenever(mockLogReader.readNextBatch()).doReturn(batch, null)
+        whenever(mockLogUploader.upload(batch.data)) doReturn status
+
+        val result = testedWorker.doWork()
+
+        verify(mockLogReader, never()).dropBatch(batch.id)
+        assertThat(result)
+            .isEqualTo(ListenableWorker.Result.failure())
+    }
+
+    @Test
+    fun `doWork mutliple batches all Success`(
+        @Forgery batches: List<Batch>
+    ) {
+        val firstBatch = batches.first()
+        val otherBatchesThenNull = Array(batches.size) {
+            batches.getOrNull(it + 1)
```

Comment:
`val batchesThenNull = Array<Log>(batches.size+1){
        batches.getOrNull(it)           
  }
`

Reply:
Nope, we need the batches except the first one, to pass to the mock below.